doc/io_streams.rdoc

```diff
@@ -0,0 +1,787 @@
+and strings written are converted from internal to external encoding.
+For further details about transcoding input and output,
+see {Encodings}[rdoc-ref:io_streams.rdoc@Encodings].
+
+If the external encoding is <tt>'BOM|UTF-8'</tt>, <tt>'BOM|UTF-16LE'</tt>
+or <tt>'BOM|UTF16-BE'</tt>,
+Ruby checks for a Unicode BOM in the input document
+to help determine the encoding.
+For UTF-16 encodings the file open mode must be binary.
+If the BOM is found,
+it is stripped and the external encoding from the BOM is used.
+
+Note that the BOM-style encoding option is case insensitive,
+so <tt>‘bom|utf-8’</tt> is also valid.
+
+=== Position
+
+An \IO stream has a nonnegative integer _position_,
+which is the byte offset at which the next read or write is to occur;
+the relevant methods:
+
+- +#tell+ (aliased as #pos): Returns the current position (in bytes) in the stream:
+
+    f = File.new('t.txt')
+    f.tell # => 0
+    f.gets # => "First line\n"
+    f.tell # => 12
+    f.close
+
+- +#pos=+:  Sets the position of the stream (in bytes):
+
+    f = File.new('t.txt')
+    f.tell     # => 0
+    f.pos = 20 # => 20
+    f.tell     # => 20
+    f.close
+
+- +#seek+: Sets the position of the stream to a given integer +offset+
+  (in bytes), with respect to a given constant +whence+, which is one of:
+
+  - +:CUR+ or <tt>IO::SEEK_CUR</tt>:
+    Repositions the stream to its current position plus the given +offset+:
+
+      f = File.new('t.txt')
+      f.tell            # => 0
+      f.seek(20, :CUR)  # => 0
+      f.tell            # => 20
+      f.seek(-10, :CUR) # => 0
+      f.tell            # => 10
+      f.close
+
+  - +:END+ or <tt>IO::SEEK_END</tt>:
+    Repositions the stream to its end plus the given +offset+:
+
+      f = File.new('t.txt')
+      f.tell            # => 0
+      f.seek(0, :END)   # => 0  # Repositions to stream end.
+      f.tell            # => 52
+      f.seek(-20, :END) # => 0
+      f.tell            # => 32
+      f.seek(-40, :END) # => 0
+      f.tell            # => 12
+      f.close
+
+  - +:SET+ or <tt>IO:SEEK_SET</tt>:
+    Repositions the stream to the given +offset+:
+
+      f = File.new('t.txt')
+      f.tell            # => 0
+      f.seek(20, :SET) # => 0
+      f.tell           # => 20
+      f.seek(40, :SET) # => 0
+      f.tell           # => 40
+      f.close
+
+- +#rewind+: Positions the stream to the beginning:
+
+    f = File.new('t.txt')
+    f.tell     # => 0
+    f.gets     # => "First line\n"
+    f.tell     # => 12
+    f.rewind   # => 0
+    f.tell     # => 0
+    f.close
+
+=== Lines
+
+Some reader methods in \IO streams are line-oriented;
+such a method reads one or more lines,
+which are separated by an implicit or explicit line separator.
+
+These methods are included (except as noted) in classes Kernel, IO, File,
+and {ARGF}[rdoc-ref:ARGF]:
+
+- +#each_line+ - passes each line to the block; not in Kernel:
+
+    f = File.new('t.txt')
+    f.each_line {|line| p line }
```

Comment:
maybe is a good place to use `open` with a block
```suggestion
   File.open('t.txt') do |f|
     f.each_line {|line| p line }
   end
```

Reply:
I have other plans for File.open.